Objects/longobject.c

```diff
@@ -167,20 +170,34 @@ _PyLong_Copy(PyLongObject *src)
     return (PyObject *)result;
 }
 
-/* Create a new int object from a C long int */
+static PyObject *
```

Comment:
Please add a comment to explain that x must not be a small integer and must be in the range [-PyLong_BASE; PyLong_BASE].

I suggest to rename the function to _PyLong_FromMedium, since it's private (static).